samples/bluetooth/peripheral_hr_coded/src/main.c

```diff
@@ -0,0 +1,197 @@
+/* main.c - Application main entry point */
+
+/*
+ * Copyright (c) 2015-2016 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <string.h>
+#include <errno.h>
+#include <sys/printk.h>
+#include <zephyr.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/uuid.h>
+#include <bluetooth/gatt.h>
+#include <bluetooth/services/bas.h>
+#include <bluetooth/services/hrs.h>
+
+static struct k_work start_advertising_worker;
+
+static struct bt_le_ext_adv *adv;
+static struct bt_conn *default_conn;
+
+static const struct bt_data ad[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+	BT_DATA_BYTES(BT_DATA_UUID16_ALL, 0x0d, 0x18, 0x0f, 0x18, 0x0a, 0x18),
+};
+
+static void connected(struct bt_conn *conn, uint8_t conn_err)
+{
+	int err;
+	struct bt_conn_info info;
+	const struct bt_conn_le_phy_info *phy_info;
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (conn_err) {
+		printk("Connection failed (err %d)\n", conn_err);
+		return;
+	}
+
+	default_conn = bt_conn_ref(conn);
+
+	err = bt_conn_get_info(conn, &info);
+	if (err) {
+		printk("Failed to get connection info\n");
+	}
+	phy_info = info.le.phy;
+
+	printk("Connected: %s, tx_phy %u, rx_phy %u\n",
+			addr, phy_info->tx_phy, phy_info->rx_phy);
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	printk("Disconnected (reason 0x%02x)\n", reason);
+
+	if (default_conn) {
+		bt_conn_unref(default_conn);
+		default_conn = NULL;
+	}
+
+	k_work_submit(&start_advertising_worker);
+}
+
+static struct bt_conn_cb conn_callbacks = {
+	.connected = connected,
+	.disconnected = disconnected,
+};
+
+static int create_advertising_coded(void)
+{
+	int err;
+	struct bt_le_adv_param param = {};
+
+	param.options  = BT_LE_ADV_OPT_CONNECTABLE;
+	param.options |= BT_LE_ADV_OPT_EXT_ADV;
+	param.options |= BT_LE_ADV_OPT_CODED;
+	param.id = BT_ID_DEFAULT;
+	param.sid = 0;
+	param.interval_min = BT_GAP_ADV_FAST_INT_MIN_2;
+	param.interval_max = BT_GAP_ADV_FAST_INT_MAX_2;
+
+	err = bt_le_ext_adv_create(&param, NULL, &adv);
+	if (err) {
+		printk("Failed to create advertiser set (%d)\n", err);
+		return -ENOEXEC;
+	}
+
+	printk("Created adv: %p\n", adv);
+
+	err = bt_le_ext_adv_set_data(adv, ad, ARRAY_SIZE(ad), NULL, 0);
+	if (err) {
+		printk("Failed to set advertising data (%d)\n", err);
+		return -ENOEXEC;
+	}
+
+	return 0;
+}
+
+static void start_advertising_coded(struct k_work *item)
+{
+	int err;
+	uint8_t adv_index;
+
+	struct bt_le_ext_adv_start_param start_param = {
+		.timeout    = 0,
+		.num_events = 0
+	};
```

Comment:
Since both of these are set to zero you don't need this param, consider calling  `bt_le_ext_adv_start(adv, NULL);` instead.